### ISSUE TITLE
Support for attack roll formulas without a "1d20"

### ISSUE THREAD

**Issue by Stendarpaval** [RedReign/FoundryVTT-BetterRolls5e#234]
I'd like to be able to roll items with a manually determined attack roll result. It's something I need for my Mob Attack Tool module. 

This was possible on Better Rolls v1.3.10, with code like this:
```js
let mobAttackRoll = BetterRolls.rollItem(weapons[key], {},
    [
        ["header"],
        ["desc"],
        ["attack", {triggersCrit: false, isCrit: false, formula: ("0d0 + " + targetAC)}]
    ]
);
// Add damage fields from each successful hit to the same card
for (let i = 0; i < numHitAttacks; i++) {
    await mobAttackRoll.addField(["damage",{index: "all"}]);
}
await mobAttackRoll.addField(["ammo"]);
await mobAttackRoll.toMessage();
```

This produces errors in v1.3.11, specifically the `attack` field's formula: `"0d0 + " + targetAC`. 

Is there a way to set a numeric value for this formula? That'd be really awesome! 

Edit: I forgot to mention, but apparently formulas like `"1d1 + " targetAC - 1` aren't allowed either, nor something like `targetAC + "d1"`. Either of these would technically solve my issue, in a roundabout way.

**Comment by CarlosFdez** [comment]
So what happened between v1.3.10 and v1.3.11 is that for the sake of midi compatibility, attacks now extract advantage/disadvantage as well as number of rolls from the formula. You did something....definitely weird for 5e to be doing.

I'll make it handle stuff without a d20 term, however, I think the best way to handle the above might be instead to do something like `["text", { text: HTML GO HERE }]`. That way you don't get the weirdness like clicking on the ATTACK field to expand the menu.

**Comment by CarlosFdez** [comment]
That said, update and see if it works.

**Comment by Stendarpaval** [comment]
It works, thank you so much! 

It is indeed a strange thing I'm doing, but it also relates to midi compatibility. Without an `["attack", {formula: "some dice formula"}]` field, midi-qol always registers such item rolls to miss the target. Successful mob attacks (for my Mob Attack Tool module) are supposed to always hit though, so I needed a way to to guarantee a hit. Setting the attack roll to the target's AC is one way to do that.

There's probably a better way to do this with MidiQOL.DamageOnlyWorkflow, but what I really liked about using BetterRolls.rollItem is that midi correctly interprets damage types for the purpose of resistances and immunities. I haven't been able to get that to work reliably (yet) with the workflow.

At any rate, thank you for pushing out that fix! Better Rolls was the first module I installed back when I got Foundry (because my players were used to Roll20's dual rolls) and it's fun to delve into its code now that I've learned a few things about Javascript. I've learned a lot from it, and by extension from you, so thanks!
